src/options/mkoptions.py

```diff
@@ -210,6 +210,34 @@ def get_holder_ref_decls(modules):
     return concat_format('  options::Holder{id_cap}& {id};', modules)
 
 
+def get_handler(option):
+    """Render handler call for assignment functions"""
+    optname = option.long_name if option.long else ""
+    if option.handler:
+        if option.type == 'void':
+            return 'opts.handler().{}("{}", option)'.format(option.handler, optname)
+        else:
+            return 'opts.handler().{}("{}", option, optionarg)'.format(option.handler, optname)
+    elif option.mode:
+        return 'stringTo{}(optionarg)'.format(option.type)
+    elif option.type != 'bool':
+        return 'handleOption<{}>("{}", option, optionarg)'.format(option.type, optname)
+    return None
+
+
+def get_predicates(option):
+    """Render predicate calls for assignment functions"""
+    if not option.predicates:
+        return []
+    optname = option.long_name if option.long else ""
+    if option.type == 'bool':
+        return ['opts.handler().{}("{}", option, value);'.format(x, optname) \
+                for x in option.predicates]
+    else:
+        assert option.type != 'void'
+        return ['opts.handler().{}("{}", option, value);'.format(x, optname) \
+                for x in option.predicates]
+
```

Comment:
What's the difference of these two returns?

Reply:
I guess they used to be different at some point in the past. I'll merge them.